src/miner.cpp

```diff
@@ -159,43 +162,63 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
-    coinbaseTx.vout[0].nValue = workState.nFees + GetBlockSubsidy(nHeight, chainparams.GetConsensus());
+    coinbaseTx.vout[0].nValue = winner->nFees + GetBlockSubsidy(nHeight, chainparams.GetConsensus());
     coinbaseTx.vin[0].scriptSig = CScript() << nHeight << OP_0;
-    workState.pblock->vtx[0] = MakeTransactionRef(std::move(coinbaseTx));
-    workState.pblocktemplate->vchCoinbaseCommitment = GenerateCoinbaseCommitment(*workState.pblock, pindexPrev, chainparams.GetConsensus());
-    workState.pblocktemplate->vTxFees[0] = -workState.nFees;
+    winner->pblock->vtx[0] = MakeTransactionRef(std::move(coinbaseTx));
+    winner->pblocktemplate->vchCoinbaseCommitment = GenerateCoinbaseCommitment(*winner->pblock, pindexPrev, chainparams.GetConsensus());
+    winner->pblocktemplate->vTxFees[0] = -winner->nFees;
 
-    uint64_t nSerializeSize = GetSerializeSize(*workState.pblock, SER_NETWORK, PROTOCOL_VERSION);
-    LogPrintf("CreateNewBlock(): total size: %u block weight: %u txs: %u fees: %ld sigops %d\n", nSerializeSize, GetBlockWeight(*workState.pblock), workState.nBlockTx, workState.nFees, workState.nBlockSigOpsCost);
+    uint64_t nSerializeSize = GetSerializeSize(*winner->pblock, SER_NETWORK, PROTOCOL_VERSION);
+    LogPrintf("CreateNewBlock(): total size: %u block weight: %u txs: %u fees: %ld sigops %d\n", nSerializeSize, GetBlockWeight(*winner->pblock), winner->nBlockTx, winner->nFees, winner->nBlockSigOpsCost);
 
     // Fill in header
-    workState.pblock->hashPrevBlock  = pindexPrev->GetBlockHash();
-    UpdateTime(workState.pblock, chainparams.GetConsensus(), pindexPrev);
-    workState.pblock->nBits          = GetNextWorkRequired(pindexPrev, workState.pblock, chainparams.GetConsensus());
-    workState.pblock->nNonce         = 0;
-    workState.pblocktemplate->vTxSigOpsCost[0] = WITNESS_SCALE_FACTOR * GetLegacySigOpCount(*workState.pblock->vtx[0]);
+    winner->pblock->hashPrevBlock  = pindexPrev->GetBlockHash();
+    UpdateTime(winner->pblock, chainparams.GetConsensus(), pindexPrev);
+    winner->pblock->nBits          = GetNextWorkRequired(pindexPrev, winner->pblock, chainparams.GetConsensus());
+    winner->pblock->nNonce         = 0;
+    winner->pblocktemplate->vTxSigOpsCost[0] = WITNESS_SCALE_FACTOR * GetLegacySigOpCount(*winner->pblock->vtx[0]);
 
     CValidationState state;
-    if (!TestBlockValidity(state, chainparams, *workState.pblock, pindexPrev, false, false)) {
+    if (!TestBlockValidity(state, chainparams, *winner->pblock, pindexPrev, false, false)) {
         throw std::runtime_error(strprintf("%s: TestBlockValidity failed: %s", __func__, FormatStateMessage(state)));
     }
     int64_t nTime2 = GetTimeMicros();
 
     LogPrint(BCLog::BENCH, "CreateNewBlock() packages: %.2fms (%d packages, %d updated descendants), validity: %.2fms (total %.2fms)\n", 0.001 * (nTime1 - nTimeStart), nPackagesSelected, nDescendantsUpdated, 0.001 * (nTime2 - nTime1), 0.001 * (nTime2 - nTimeStart));
```

Comment:
My thought was that `nPackagesSelected` and `nDescendantsUpdated` are mostly here to help parameterize the runtime, as we expect this to be slower the more packages are looked at and the more descendants that have to be walked to update scores.  

I guess it can be a little confusing if someone might try to interpret this as the actual number of packages in the block, because that will be incorrect after this PR...